lib/rules/no-promise-executor-return.js

```diff
@@ -59,6 +60,68 @@ function isPromiseExecutor(node, scope) {
         isGlobalReference(parent.callee, getOuterScope(scope));
 }
 
+/**
+ * Checks if the given node is a void expression.
+ * @param {ASTNode} node The node to check.
+ * @returns {boolean} - `true` if the node is a void expression
+ */
+function expressionIsVoid(node) {
+    return node.type === "UnaryExpression" && node.operator === "void";
+}
+
+/**
+ * Fixes the linting error by prepending "void " to the given node
+ * @param {Object} sourceCode context given by context.sourceCode
+ * @param {ASTNode} node The node to fix.
+ * @param {Object} fixer The fixer object provided by ESLint.
+ * @returns {Array<Object>|Object} - An array of fix objects or fix to apply to the node.
+ */
+function voidPrependFixer(sourceCode, node, fixer) {
+
+    /*
+     * prepending `void ` will fail if the node has a lower precedence than void
+     */
+    if (astUtils.getPrecedence(node) < astUtils.getPrecedence({ type: "UnaryExpression", operator: "void" })) {
+
+        // check if there are parentheses around the node to avoid redundant parentheses
+        if (!(sourceCode.getTokenBefore(node).value === "(" && sourceCode.getTokenAfter(node).value === ")")) {
+            return [
+                fixer.insertTextBefore(node, "void ("),
+                fixer.insertTextAfter(node, ")")
+            ];
+        }
+    }
+
+    if (node.parent.type === "ArrowFunctionExpression") {
+        const arrowToken = sourceCode.getTokenBefore(node, astUtils.isArrowToken);
+        const firstToken = sourceCode.getTokenAfter(arrowToken);
+
+        return fixer.insertTextBefore(firstToken, "void ");
+    }
+
+    return fixer.insertTextBefore(node, "void ");
```

Comment:
In the following example, this fix would produce invalid code:

```js
/* eslint no-promise-executor-return: ["error", { "allowVoid": true }] */

new Promise(resolve => {
    return (
        foo = resolve(1)
    );
});


```

After the "prepend void" suggestion is applied:

```js
/* eslint no-promise-executor-return: ["error", { "allowVoid": true }] */

new Promise(resolve => {
    return (
        void foo = resolve(1) // SyntaxError: Invalid left-hand side in assignment
    );
});

```

Reply:
Thanks for pointing this out. I find myself "pulling" the promise resolver out like `new Promise(r => f = r)` a lot, so this definitely helps

Reply:
The correct fix would be to insert `void ` before the left parentheses (before the first token after `return`):

```js
/* eslint no-promise-executor-return: ["error", { "allowVoid": true }] */

new Promise(resolve => {
    return void (
        foo = resolve(1)
    );
});
```